### PR TITLE
Added SMB Multichannel protocol

### PR DESCRIPTION
**Pull request by tasherif-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/_models.py [hunk 1]
```diff
@@ -134,6 +135,16 @@ def _from_generated(cls, generated):
         )
 
 
+class ProtocolProperties(GeneratedProtocolSettings):
+    def __init__(self, smb_settings=None):
+        self.smb_settings = smb_settings
+
+    @classmethod
+    def _from_generated(cls, generated):
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-file-share/azure/storage/fileshare/_models.py:143]: where is this used?

Comment (tasherif-msft) [reply sdk/storage/azure-storage-file-share/azure/storage/fileshare/_models.py:147]: In the `service_properties_deserialize` (recent commit)

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/_models.py [hunk 2]
```diff
@@ -134,6 +135,16 @@ def _from_generated(cls, generated):
         )
 
 
+class ProtocolProperties(GeneratedProtocolSettings):
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-file-share/azure/storage/fileshare/_models.py:138]: we should make the customer's life as easy as possible. 
With ProtocolProperties, they have to create SmbSettings and SmbMultichannel. While the actually thing they are setting is only a boolean to indicate If SMB multichannel is enabled. Exposing ProtocolProperties will force them to create several instances in order to setting a boolean value. 
So what we want to do is just give them an option like is_smb_multichannel_enabled (not sure if this name is good enough)
https://github.com/tasherif-msft/azure-sdk-for-python/blob/smb-multichannel/sdk/storage/azure-storage-file-share/azure/storage/fileshare/_generated/models/_models.py#L845-L861

Comment (tasherif-msft) [reply sdk/storage/azure-storage-file-share/azure/storage/fileshare/_models.py:138]: As per the discussions earlier today we will maintain a layered approach for the Protocol Settings and remove the `settings` suffix.

**Review by kasobol-msft — APPROVED** [review]

**Pull request merged**